### ISSUE TITLE
this.db.readIndex is not a function error

### ISSUE THREAD

**Issue by rubickCube** [glayzzle/php-reflection#19]
I'm trying to make this work, but I get the error message shown below, 
db.readIndex  seams not implemented...I missed something?

 ```js
var reflection = require('php-reflection');
var workspace = new reflection.Repository('/srv/develop/Models', {
    exclude: ['.git', '.svn'],
    ext: [
        '*.php','*.php3','*.php5','*.phtml',
        '*.inc','*.class','*.req'
    ],  
    scanVars: true,
    scanExpr: true,
    encoding: 'utf8',
    cacheByFileSize: true
});


var workers = [ 
    workspace.parse('ClassA.php'),
    workspace.parse('ClassB.php'),
    workspace.parse('ClassC.php')
];
Promise.all(workers).then(function() {
    console.log('-- list of functions :');
    workspace.getByType('function').each(function(fn) {
        console.log('Function Name : ', fn.name);
        console.log('Located into : ', fn.getFile().name);
        console.log('At line : ', fn.position.start.line);
    }); 
}).catch(function(error){
    
 console.log(error);   
    
});

```
Console log:
> -- list of functions :
> TypeError: this.db.readIndex is not a function
>     at repository.getByType (/srv/develop/testing/PhpNodeParser/node_modules/php-reflection/src/repository.js:147:13)
>     at /srv/develop/testing/PhpNodeParser/example.js:22:15
>     at \<anonymous\>

**Comment by b4dnewz** [comment]
Hi @ichiriac, since we was talking in the other repository PR I take this opportunity to say I would like to use this project for building (in my spare time) a static code vulnerability analisis tool and since I've noticed months ago when I first tried it it was broken I would like to help here too.

I've tried a previous version in the past which use a previous version of [grafine](https://github.com/glayzzle/grafine) and it was working quite good, but I found some problems to understand how to use it properly for the lack of documentation, maybe once I dig deeper I can write some markdown files with examples and stuff.

Do you have any advice about the error or anything else? 

Thanks in advance!

**Comment by ichiriac** [comment]
Hi,

I need to take some time to work again on this project, the actual master version may be a work in progress on a rewrite/refactoring so that's why it's not stable.

If you can we can chat on [gitter](https://gitter.im/glayzzle/Lobby) in about 30 minutes / we can talk about how it was implemented and what is not finished yet.

Also take a look at this project : https://github.com/glayzzle/code-sniper - it's almost ready (about 1 day of work to be done) and could be a good start for a static analyzer tool

**Comment by b4dnewz** [comment]
@ichiriac sorry for the late response I took the dog ou.

awesome, I didnt see code-sniper, unfortunately I can't be on gitter chat right now because I'm going out for lunch and I need to leave right now, but I will write my questions maybe later or If you want another time we can chat.

Have a good day!

**Comment by ichiriac** [comment]
okay, no problem, when you want, I'll be online

**Comment by iliyaZelenko** [comment]
Same problem.

**Comment by DanStevens** [comment]
I'm getting this error The change in patch 82bec46  doesn't seem to be present in the npm package.

**Comment by ichiriac** [comment]
Hi @DanStevens,

Right, I've did not released it on npm since 2 years - I'll make a release soon

**Comment by ichiriac** [comment]
it's released now in version 2.3.1
